### ISSUE TITLE
[AOTInductor] support specify outputs which should be captured

### ISSUE THREAD

**Issue by zzq96** [pytorch/pytorch#149036]
### 🚀 The feature, motivation and pitch

In train, model forward may return more outputs for computing loss, like `return {"logits":logits, "rpr":rpr}` 

but in inference, we only need  some of them, like `return {"logits":logits}`, so torch can simplify graph and ignore some nodes related to `rpr`.

cc @desertfire @penguinwu @yushangdi @benjaminglass1 @jataylo @iupaikov-amd @chauhang @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi @suo @ydwu4 @chenyang78 @zou3519 @bdhirsh
